### PR TITLE
GEOMESA-1192 Fix KafkaDataStore's getTypeNames method

### PR DESCRIPTION
**Pull request by Afterglow375**
Signed-off-by: Alex Tatusko <atatusko@ccri.com>

### CODE REVIEW

File: geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala [hunk 1]
```diff
@@ -112,9 +112,11 @@ trait KafkaDataStoreSchemaManager extends DataStore with LazyLogging {
     }.getOrElse(None)
   }
 
-  override def getNames: util.List[Name] =
-    zkUtils.zkClient.getChildren(zkPath).asScala.map(name => new NameImpl(name) : Name).asJava
-
+  override def getNames: util.List[Name] = {
+    zkUtils.zkClient.getChildren(zkPath).asScala
+      .filter(name => zkUtils.zkClient.exists(getTopicPath(name)))
```
Comment (cne1x) [inline geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala]: Do we have (or should we have) any concerns about dead topic names accumulating in Zookeeper?

Comment (Afterglow375) [inline geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala]: Dead topic names accumulating in zookeeper would depend on the user. removeSchema will remove the zk nodes pertaining to the schema and calling createSchema with a zkpath + sft name that already exists will throw an IllegalArgumentException.

Comment (cne1x) [inline geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala]: Makes sense to me.  Thanks!

**Comment by jnh5y** [comment]
I'm out of the loop on this one.  What is the issue being addressed?

**Comment by elahrvivaz** [comment]
merging

**Pull request merged**